applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -29,7 +48,62 @@ impl wallet_server::Wallet for WalletGrpcServer {
             Ok(resp) => Ok(Response::new(GetCoinbaseResponse {
                 transaction: Some(resp.into()),
             })),
-            Err(_err) => unimplemented!(),
+            Err(err) => Err(Status::unknown(err.to_string())),
         }
     }
+
+    async fn transfer(&self, request: Request<TransferRequest>) -> Result<Response<TransferResponse>, Status> {
+        let message = request.into_inner();
+        let recipients = message
+            .recipients
+            .into_iter()
+            .enumerate()
+            .map(|(idx, dest)| -> Result<_, String> {
+                let pk = CommsPublicKey::from_hex(&dest.address)
+                    .map_err(|_| format!("Destination address at index {} is malformed", idx))?;
+                Ok((dest.address, pk, dest.amount, dest.fee_per_gram, dest.message))
+            })
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(Status::invalid_argument)?;
+
+        let transfers = recipients
```

Comment:
Perhaps it should try reserve all the funds before hand so that the batch executes as a unit?

Reply:
YesI have been using the term "encumber" to research outputs for a given transaction. Currently the only way it happens is to initiate a transaction via the transaction service so some logic will need to be added to both services to reserve funds for a batch of transactions without starting the transactions.

Reply:
Perhaps that is out of scope for this, and can be added as a TODO or issue